release/serve_tests/workloads/anyscale_service_utils.py

```diff
@@ -0,0 +1,89 @@
+from contextlib import contextmanager
+import logging
+import os
+from typing import Dict, List, Union
+
+from anyscale import service
+import ray
+from ray._private.test_utils import wait_for_condition
+from ray.serve._private.utils import get_random_string
+
+
+logger = logging.getLogger(__file__)
+logging.basicConfig(level=logging.INFO)
+
+DEFAULT_CLUSTER_ENV = "default_cluster_env_nightly-ml_py39"
+
+
+def check_service_state(service_name: str, expected_state: str):
+    state = service.status(name=service_name).state
+    logger.info(
+        f"Waiting for service {service_name} to be {expected_state}, currently {state}"
+    )
+    assert (
+        str(state) == expected_state
+    ), f"Service {service_name} is {state}, expected {expected_state}."
+    return True
+
+
+@contextmanager
+def start_service(
+    service_name: str,
+    compute_config: Union[str, Dict],
+    applications: List[Dict],
+    add_unique_suffix: bool = True,
+):
+    """Starts an Anyscale Service with the specified configs.
+
+    Args:
+        service_name: Name of the Anyscale Service. The actual service
+            name may be modified if `add_unique_suffix` is True.
+        compute_config: The configuration for the hardware resources
+            that the cluster will utilize.
+        applications: The list of Ray Serve applications to run in the
+            service.
+        add_unique_suffix: Whether to append a unique suffix to the
+            service name.
+    """
+
+    cluster_env = os.environ.get("ANYSCALE_JOB_CLUSTER_ENV_NAME", DEFAULT_CLUSTER_ENV)
+
+    if add_unique_suffix:
+        ray_version = (
+            ray.__commit__[:8]
+            if ray.__commit__ != "{{RAY_COMMIT_SHA}}"
+            else get_random_string()
```

Comment:
probably should do both the SHA and random string, else multiple runs kicked off from the same commit have a race

Reply:
done